src/common/transformations/src/transformations/op_conversions/convert_avgpool_downgrade.cpp

```diff
@@ -0,0 +1,129 @@
+            const auto shape = std::make_shared<ov::op::v3::ShapeOf>(input, element::i64);
+            const auto gth_in_dims = std::make_shared<ov::op::v8::Gather>(shape, dim_idxs, zero);
+            const auto in_left_padded = std::make_shared<ov::op::v1::Add>(gth_in_dims, padding_begin_node);
+
+            // gather output spatial dims and prepare it for compare as values (out_dim - 1) * stride
+            const auto ap = std::make_shared<ov::op::v1::AvgPool>(input,
+                                                                  avg_pool_v14->get_strides(),
+                                                                  avg_pool_v14->get_pads_begin(),
+                                                                  avg_pool_v14->get_pads_end(),
+                                                                  avg_pool_v14->get_kernel(),
+                                                                  avg_pool_v14->get_exclude_pad(),
+                                                                  ov::op::RoundingType::CEIL);
+            const auto shape_of_ap = std::make_shared<ov::op::v3::ShapeOf>(ap, element::i64);
+            const auto gth_out_dims = std::make_shared<ov::op::v8::Gather>(shape_of_ap, dim_idxs, zero);
+            const auto out_sub_one = std::make_shared<ov::op::v1::Subtract>(gth_out_dims, one);
+            const auto stride_node = ov::op::v0::Constant::create(element::i64, Shape{strides.size()}, strides);
+            const auto out_mul_stride = std::make_shared<ov::op::v1::Multiply>(out_sub_one, stride_node);
+
+            // if (in_dim + pad) < ((out_dim - 1) * stride) sliding window in bound use end padding.
+            const auto in_gt_out = std::make_shared<ov::op::v1::Greater>(out_mul_stride, in_left_padded);
+            const auto selected_pads = std::make_shared<ov::op::v1::Select>(in_gt_out, padding_end_node, zero);
+
+            // apply padding on input clear pads attribute
+            const auto pb =
+                std::make_shared<ov::op::v0::Concat>(OutputVector{pads_remaining->output(0), padding_end_node}, 0);
+            const auto pe = std::make_shared<ov::op::v0::Concat>(OutputVector{pads_remaining, selected_pads}, 0);
+            auto minus_inf =
+                ov::op::v0::Constant::create(element::f32, Shape{}, {-std::numeric_limits<float>::infinity()})
+                    ->output(0);
+            std::shared_ptr<ov::Node> convert_like_node = std::make_shared<ov::op::v1::ConvertLike>(minus_inf, input);
+            const auto pad_node =
+                std::make_shared<ov::op::v12::Pad>(input, pb, pe, convert_like_node, op::PadMode::CONSTANT);
+            auto pads_begin = avg_pool_v14->get_pads_begin();
+            auto pads_end = avg_pool_v14->get_pads_end();
+            std::fill_n(pads_begin.begin(), pads_begin.size(), 0);
+            std::fill_n(pads_end.begin(), pads_end.size(), 0);
+
+            avg_pool_v1 = std::make_shared<ov::op::v1::AvgPool>(pad_node,
```

Comment:
I followed the subgraph: https://github.com/openvinotoolkit/openvino/blob/647ef4e701dd39adca85ae9daee56f75a67f84aa/src/frontends/pytorch/src/op/max_poolnd.cpp#L58-L102
Which itself produces a second Pooling node for its output shape and its functionality is very similar to my change in shape_inference in the Core PR.

I see that you've also linked https://github.com/openvinotoolkit/openvino/blob/647ef4e701dd39adca85ae9daee56f75a67f84aa/src/frontends/pytorch/src/op/avg_poolnd.cpp#L50-L68
I'm confused, which one of them is the workaround? I believe the first one, which is reflected in the downgrade transformations. Should the second one also be added? cc @praasz

Reply:
Those are links to the corresponding files for the PT FE MaxPool and PT FE AvgPool. 
Looks like the approach is not unified between ops, so the subgraph produced for MaxPool is not the same as for AvgPool, and it could be done the same way in the decomposition. 

As I understand, within this PR the approach from PT MaxPool is applied for both.
If correctness of such decomposition can be confirmed for AvgPool, then ok.

@praasz Could you please comment as an author of the Pytorch FE work-around?

Reply:
The FE work-around was done for max pool only, because it was reported only for this op. The `AvgPool` has similar issue but task for new operator version has been created to solve it in operators. The new ceil mode was only applied for MaxPool in Pytorch FE. The FE implementation of  `AvgPool`  not correctly calculate this ceil mode.

Regarding the 2nd pool operator, it has been used on `ShapeOf` sub-graph to get shape for ceil mode and then additional nodes are to calculate ceil mode for PyTorch and apply required padding.
This 2nd `pool` operator is used to do shape inference (no data evaluation). It probably can be replaced by other operators but then old ceil mode has to be calculated by some graph (which do same stuff as shape inference).

`ShapeOf` has been used to have correct original shape for dynamic shapes also.

Reply:
As agreed offline subgraph workaround has been removed for `AvgPool`. Since this PR moves functionality from PT FE in 1:1 fashion, we decided not to add neither dynamic shapes support nor `MaxPool`'s 2nd output fix.

Reply:
We agreed to don't introduce any subgraph modifications within this PR, and indeed move PT FE functionality 1:1.
But there is still no parity between `AvgPool` and the PT decomposition, this part is not included:
https://github.com/openvinotoolkit/openvino/blob/647ef4e701dd39adca85ae9daee56f75a67f84aa/src/frontends/pytorch/src/op/avg_poolnd.cpp#L50-L68
Is that intentional?